sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -183,9 +235,17 @@ export class EventHubClient {
    *
    * @returns {Promise<Delivery>} Promise<Delivery>
    */
-  async send(data: EventData, partitionId?: string | number): Promise<Delivery> {
+  async send(data: EventData, partitionId?: string | number): Promise<Delivery>;
+  async send(data: EventData, options?: SendOptions): Promise<Delivery>;
+  async send(data: EventData, partitionIdOrptions?: string | number | SendOptions): Promise<Delivery> {
+    let partitionId: string | number | undefined;
+    if (typeof partitionIdOrptions === "string" || typeof partitionIdOrptions === "number") {
+      partitionId = partitionIdOrptions;
+    } else if (partitionIdOrptions && partitionIdOrptions.hasOwnProperty("partitionId")) {
+      partitionId = partitionIdOrptions.partitionId;
+    }
     const sender = EventHubSender.create(this._context, partitionId);
-    return sender.send(data);
+    return typeof partitionIdOrptions === "object" ? sender.send(data, partitionIdOrptions) :  sender.send(data);
```

Comment:
We have an if/else block in the beginning. Can we re-use that to decide what the options should be instead of addition another check here?